plugins/acpi-dmar/fu-plugin-acpi-dmar.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+#include "fu-acpi-dmar.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+gboolean
+fu_plugin_add_security_attrs (FuPlugin *plugin, GPtrArray *attrs, GError **error)
+{
+	FwupdSecurityAttr *attr;
+	g_autofree gchar *fn = NULL;
+	g_autofree gchar *path = NULL;
+	g_autoptr(FuAcpiDmar) dmar = NULL;
+	g_autoptr(GBytes) blob = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* only Intel */
+	if (!fu_common_is_cpu_intel ())
+		return TRUE;
+
+	/* create attr */
+	attr = fwupd_security_attr_new ("org.uefi.ACPI.Dmar");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_THEORETICAL);
+	fwupd_security_attr_set_name (attr, "ACPI kDMAp");
+	g_ptr_array_add (attrs, attr);
+
+	/* load DMAR table */
+	path = fu_common_get_path (FU_PATH_KIND_ACPI_TABLES);
+	fn = g_build_filename (path, "DMAR", NULL);
+	blob = fu_common_get_contents_bytes (fn, &error_local);
+	if (blob == NULL) {
+		fwupd_security_attr_set_result (attr, error_local->message);
+		return TRUE;
+	}
+	dmar = fu_acpi_dmar_new (blob, error);
+	if (dmar == NULL) {
+		g_prefix_error (error, "could not parse %s: ", fn);
+		return FALSE;
+	}
+	if (fu_acpi_dmar_get_opt_in (dmar))
+		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
```

Comment:
Don't you want to explain here the failure scenario and what it means in the attribute?

Reply:
I think i'd struggle to explain the failure in less than 5 words. Got any ideas?

Reply:
Here's 4:
`Pre-boot DMA protection unavailable`

Reply:
Added, thanks.